File: Pattern 5.c
/*
Accept number of row and number of column from user and display below pattern.
Input   iRow=3   icol=5
output:
        1   1   1   1
        2   2   2   2
        3   3   3   3
        4   4   4   4
       
*/
#include<stdio.h>
void pattern(int iRow,int iCol)
{
    int no = 1;
    int i = 0,j = 0;
    for(i =1;i<=iRow;i++,no++)
    {
        for(j = 1;j<=iCol;j++)
        {
            printf("%d\t",no);
        }
    printf("\n");
    }

}
int main()
{
    int iValue1 = 0, iValue2=0;
    printf("Enter number of row and columns:\n");
    scanf("%d %d",&iValue1,&iValue2);
    pattern(iValue1,iValue2);
    return 0;
}

File: Pattern5.java
/*
    input : iRow - 4  iCol - 4
    output:

            1   2   3   4   5
            1   2           5
            1       3       5
            1           4   5
            1   2   3   4   5
 */
import java.util.*;

class Pattern
{
    public void pattern(int iRow, int iCol)
    {
        int i  = 0, j = 0;

        for(i =  1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol ; j++)
            {
                if((j == 1)||(j == iCol)||(i == iRow)||(i == 1))
                {
                    System.out.print(j+"\t");
                }
                
                else if(i == j)
                {
                    System.out.print(j+"\t");
                } 
                else
                {
                    System.out.print(" \t");
                }
            }
             System.out.println();
        }
    }
}
class Pattern5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of Rows");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of Column");
        int iCol = sobj.nextInt();

        Pattern pobj = new Pattern();
        pobj.pattern(iRow,iCol);
    }
}